src/Quadrature.jl

```diff
@@ -403,22 +403,35 @@ ZygoteRules.@adjoint function __solvebp(prob,alg,sensealg,lb,ub,p,args...;kwargs
             _f = (x) -> prob.f(dx,x,p)
             if sensealg.vjp isa ZygoteVJP
                 dfdp = function (dx,x,p)
-                    _,back = Zygote.pullback(p) do p
-                        _dx = Zygote.Buffer(x, prob.nout, size(x,2))
-                        prob.f(_dx,x,p)
-                        copy(_dx)
+                    for idx in 1:size(x,2)
+                        _,back = Zygote.pullback(p) do p
+                            _dx = Zygote.Buffer(x, prob.nout)
+                            prob.f(_dx,@view(x[:,idx]),p)
+                            copy(_dx)
+                        end
+                        dx[:,idx] .= back(y)[1]
                     end
-                    back(y)[1]
                 end
             elseif sensealg.vjp isa ReverseDiffVJP
                 error("TODO")
             end
         else
             _f = (x) -> prob.f(x,p)
             if sensealg.vjp isa ZygoteVJP
-                dfdp = function (x,p)
-                    _,back = Zygote.pullback(p->prob.f(x,p),p)
-                    back(y)[1]
+                if prob.batch > 0
+                    dfdp = function (x,p)
+                        out = zeros(length(p),size(x,2))
+                        for idx in 1:size(x,2)
+                            _,back = Zygote.pullback(p->prob.f(@view(x[:,idx]),p),p)
```

Comment:
Still feels like a hack, but the `pullback` can get taken out of the loop w/o the view and then loop through `back` accordingly. 

Using MWE:
```julia
using Zygote
w(x,p) = x[1,:]*p[1] + x[2,:]*p[2]
batch = 5
_,back  = Zygote.pullback(p->w(repeat([1,2],inner=(1,batch)),p),[3 2])

out = zeros(2, batch )
for idx in 1:batch
    z = zeros(batch)
    z[idx] = 1
    out[:,idx] = back(z)[1]
end
```

Reply:
You shouldn't need to pull back the pieces separately though: this is pullback of the identity matrix of a sparse Jacobian which is the matrix coloring problem that has a single color solution.

Reply:
I'm not sure I understand.

Reply:
```julia
using Zygote
w(x,p) = x[1,:]*p[1] + x[2,:]*p[2]
batch = 5
_,back  = Zygote.pullback(p->w(repeat([1,2],inner=(1,batch)),p),[3 2])

out = zeros(2, batch )
z = zeros(batch)
for idx in 1:batch
    z[idx] = 1
    out[:,idx] = back(z)[1]
    z[idx] = 0
end
out
```

is how I'd do it. It's not a hack but the right way to do it, because you have to pull back the basis elements if the function isn't sparse, and here it's not sparse in `p`. This keeps the primal calculation down to a single time and is the equivalent to the forwarddiff chunk version.